src/main/java/vg/civcraft/mc/civmodcore/dao/ConnectionPool.java

```diff
@@ -0,0 +1,96 @@
+package vg.civcraft.mc.civmodcore.dao;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+
+/**
+ * Handy Connection Pool / Database wrapper for use by all plugins.
+ * 
+ * @author ProgrammerDan
+ *
+ */
+public class ConnectionPool {
+	
+	private HikariDataSource datasource;
+	
+	private Logger logger;
+	
+	/**
+	 * Creates the Database connection pool backed by HikariCP.
+	 * 
+	 * @param log The logger to use
+	 * @param user The user to connection as
+	 * @param pass The password to use
+	 * @param host The host to connect to
+	 * @param port The port on the host to connect to
+	 * @param database The database to use
+	 * @param poolSize The maximum size of the connection pool (< 10 recommended)
+	 * @param connectionTimeout The longest a query can run until timeout occurs (1-5s recommended)
+	 * @param idleTimeout The longest a connection can sit idle before recycling (10min recom. check dbms config)
+	 * @param maxLifetime The absolute longest length of time a connection can exist (2hr recom. check dbms config)
+	 */
+	public ConnectionPool(Logger log, String user, String pass, String host, int port, String database,
```

Comment:
Plan is to convert it to a ConfigurationSerializable; a builder or factory might be fun I guess, would be easier to expose more features of the connection pool that way.